pkg/deployment/all_in_one.go

```diff
@@ -63,15 +63,17 @@ func (a *AllInOne) Get() *appsv1.Deployment {
 
 	configmap.Update(a.jaeger, commonSpec, &options)
 	sampling.Update(a.jaeger, commonSpec, &options)
-	tls.Update(a.jaeger, commonSpec, &options)
-	ca.Update(a.jaeger, commonSpec)
-	ca.AddServiceCA(a.jaeger, commonSpec)
+	if len(util.FindItem("--collector.grpc.tls.enabled", options)) == 0 {
+		tls.Update(a.jaeger, commonSpec, &options)
```

Comment:
This doesn't feel right: if the option is set, then tls.Update will be skipped (and ca, and ca.AddServiceCA). Why did we have this as part of all reconciliation loops before? What happens when on the first reconciliation we did not have this parameter, causing the operator to add `--reporter.grpc.tls.enabled=true`, and the user then manually sets this to `false`?

Reply:
Good point, I need to test this case, 

I ran manually tests only for the first case, I'll check what happens in the case of an update to the CR. but it seems like you're right.

Reply:
Did you test this case?

Reply:
I tested it with the last changes I did, it seems to work fine.